### PR TITLE
Support Posix based Waiter APIs for SB16 and tests

### PR DESCRIPTION
**Pull request by maxz-lab**
Introduced API SbPosixSocketWaiterAdd, SbPosixSocketWaiterRemove and
  callback SbPosixSocketWaiterCallback.
Tests added are
  SbPosixSocketWaiterAdd
  SbPosixSocketWaiterRemove
  SbPosixSocketWaiterWait
  SbPosixSocketWaiterWaitTimed
libevent and win32 SocketWaiter module supports both integer based socket and
  SbSocket socket in SB16.

b/302741384

### CODE REVIEW

File: starboard/shared/libevent/socket_waiter_add.cc [hunk 1]
```diff
@@ -45,3 +46,35 @@ bool SbSocketWaiterAdd(SbSocketWaiter waiter,
 
   return waiter->Add(socket, context, callback, interests, persistent);
 }
+
+#if SB_API_VERSION >= 16
+
+bool SbPosixSocketWaiterAdd(SbSocketWaiter waiter,
```
Comment (haozheng-cobalt) [inline starboard/shared/libevent/socket_waiter_add.cc:52]: Are we not going to deprecate SbSocketWaiter and SbPosixSocketWaiterCallback for SB 16?

Comment (maxz-lab) [reply starboard/shared/libevent/socket_waiter_add.cc:52]: This change intents to add the POSIX APIs. 
The deprecation will be applied by future CL.

File: starboard/CHANGELOG.md [hunk 2]
```diff
@@ -9,6 +9,11 @@ since the version previous to it.
 
 ## Version 16
 
+### Implement SbPosixSocketWaiter APIs with tests
```
Comment (y4vor) [inline starboard/CHANGELOG.md:12]: You don't need this. You already added the API.

File: starboard/CHANGELOG.md [hunk 3]
```diff
@@ -19,6 +24,7 @@ APIs.
 
 ## Deprecated `Accessibility` header
 
+### Deprecated `Accessibility` header
```
Comment (y4vor) [inline starboard/CHANGELOG.md:27]: this is dupe.

File: starboard/shared/win32/posix_emu/include/sys/socket.h [hunk 4]
```diff
@@ -69,6 +69,9 @@ int sb_setsockopt(int socket,
                   int option_len);
 #define setsockopt sb_setsockopt
 
+int posixSocketGetFdFromHandle(SOCKET socket);
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/include/sys/socket.h:72]: use lower case with underscores?

File: starboard/shared/win32/posix_emu/include/sys/socket.h [hunk 5]
```diff
@@ -69,6 +69,9 @@ int sb_setsockopt(int socket,
...
+SOCKET posixSocketGetHandleFromFd(int socket);
```
Comment (y4vor) [inline starboard/shared/win32/posix_emu/include/sys/socket.h:73]: same

**Review by y4vor — APPROVED** [review]

**Pull request merged**